server/models.py

```diff
@@ -6,7 +6,12 @@
 from django.contrib.auth.models import User as DJUser
 
 import sys, datetime, logging
-
+#Vending Machine Stock
+class Venditem(models.Model):
```

Comment:
I think stock should be a separate object. Why? because you could imagine a scenario where you have multiple machines and you want to track stock in them individually. Also, you might want to keep DB records for things you're not currently stocking.

What I'd propose is something like this:

```
class VendItem(models.Model):
    name = ...
    price = ...
    
class MachineItem(models.Model):
    item = models.ForeignKey(VendItem)
    tool = models.ForeignKey(Tool)
    position = models.PositiveIntegerField()
    stock = models.PositiveIntegerField()

```

this way you can separately track items you might want to stock, and items you actually have in the machine

Reply:
The classes look like this now
```
class VendItem(models.Model):
  name = models.TextField(default = '')
  price = models.PositiveIntegerField(default = 0) # Price in pence

class MachineItem(models.Model):
    item = models.ForeignKey(VendItem)
    tool = models.ForeignKey(Tool)
    position = models.PositiveIntegerField()
    stock = models.PositiveIntegerField()
```